wellcomeml/ml/keras_vectorizer.py

```diff
@@ -21,20 +26,58 @@ def transform(self, X, *_):
         sequences = self.tokenizer.texts_to_sequences(X)
         return pad_sequences(sequences, maxlen=self.sequence_length)
 
-    def build_embedding_matrix(self, embeddings_path):
-        embeddings_index = {}
-        with open(embeddings_path) as f:
-            for line in f:
-                word, coefs = line.split(maxsplit=1)
-                coefs = np.fromstring(coefs, "f", sep=" ")
-                embeddings_index[word] = coefs
+    def build_embedding_matrix(self, embeddings_path=None, word_vectors=None):
+        """
+        Builds an embedding matrix from either a local embeddings path
+        or a gensim pre-trained word vector path
+
+        Args:
+            embeddings_path: A local directory to word embeddings
+            word_vectors: The name of a GenSim pre-trained word vector model
+                e.g. 'glove-twitter-25', for the complete list:
+                https://github.com/RaRe-Technologies/gensim-data#models
+
+        Returns:
+            An embedding matrix
+
+        """
+        if embeddings_path:
+            if path.exists(embeddings_path):
+                embeddings_index = {}
+                with open(embeddings_path) as f:
+                    for line in f:
+                        word, coefs = line.split(maxsplit=1)
+                        coefs = np.fromstring(coefs, "f", sep=" ")
+                        embeddings_index[word] = coefs
+                    emb_dim = len(coefs)
+            else:
+                logger.error("Incorrect local embeddings path")
```

Comment:
should we log an error or throw an exception. does the pipeline work if you get None back?

Reply:
yeh i think youre right it should really throw an exception and stop